### PR TITLE
fix(bitnami): use a different comparer for detecting vulnerabilities

### PR DESCRIPTION
**Pull request by juan131**
## Description

As it's explained at https://github.com/aquasecurity/trivy/discussions/5622, Bitnami doesn't strictly follow semver versioning since a revision is added to versions using what [semver spec](https://semver.org/#spec-item-9) consider "pre-releases". Sth similar is done by Debian, see:

- https://manpages.debian.org/wheezy/dpkg-dev/deb-version.5.en.html

This PR adds a new comparer on "detector" pkg for Bitnami based on [Bitnami's go-version](https://github.com/bitnami/go-version) so Bitnami versions including revisions are properly managed.

> Note: refer to "Reproduction Steps" in the original discussion for a "before" and "after" example.

## Related issues

- Close https://github.com/aquasecurity/trivy/discussions/5622

## Checklist

- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [x] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by knqyf263** [comment]
Oh, I completely forgot about the constraints. Thanks for the heads-up. I think that logic is too much in Trivy, then. Can we create a new repository, like go-bitnami-version? We can build it on top of `go-deb-version`. For example, [go-npm-version](https://github.com/aquasecurity/go-npm-version) uses go-version as it follows semver, and implements the constraints only.
We can do that under `aquasecurity/` or `bitnami/`. Which do you prefer?

**Comment by juan131** [comment]
@knqyf263 the repository has been created and the PR has been adapted to use it:

- https://github.com/bitnami/go-version

**Comment by knqyf263** [comment]
I am currently on vacation due to moving to a new country. I hope to find some time to review, but I don't have much time available. I apologize for any inconvenience caused.

File: go.mod [hunk 1]
```diff
@@ -115,6 +115,8 @@ require (
 	modernc.org/sqlite v1.23.1
 )
 
+require github.com/bitnami/go-version v0.0.0-20231130082345-3f6796aa059b
```
Comment (DmitriyLewen) [inline go.mod:118]: This is not the latest version: https://github.com/bitnami/go-version/commit/bb00604d650cf333cc3fcbc195983a1c3d73233e
This is fine?

**Review by DmitriyLewen — COMMENTED** [review]
LGTM
Left 1 comment

@juan131 take a look, please

**Review by DmitriyLewen — APPROVED** [review]
LGTM
@knqyf263 I think we can merge this PR.

**Review by knqyf263 — APPROVED** [review]
Thanks! I changed it to a bug fix so we can include it in v0.48.1.

**Comment by juan131** [comment]
Thanks so much!

**Pull request merged**